package com.gautam;

public class ExceptionHandler {

	public static void demonstrateCustomException() throws CustomException {
		throw new CustomException("This is a custom exception message");
	}

	public static void demonstrateExceptionHierarchy() throws CustomException {
		try {
			demonstrateCustomException();
		} catch (CustomException e) {
			System.out.println("Caught CustomException: " + e);
			throw e; // rethrowing the exception
		}
	}

	public static void main(String[] args) {
		try {
			demonstrateExceptionHierarchy();
		} catch (CustomException e) {
			System.out.println("Exception handled in main: " + e);
		} finally {
			System.out.println("Finally block executed");
		}
	}
}
